### DISCUSSION TITLE
current_user: what's the difference between optional and active?

### DISCUSSION THREAD

**Discussion by kartzke** [fastapi-users/fastapi-users#1034] [Q&A] [answered]
Hi, can someone tell me what the difference is between these two parameters?  The documentation isn't clear for me... if a user is logged in, what makes them "active" vs "inactive" ?  

When should I used this: 
`current_active_user = fastapi_users.current_user(active=True)`

And when should I use this: 
`current_user = fastapi_users.current_user()`

[From the documentation](https://fastapi-users.github.io/fastapi-users/10.1/usage/current-user/?h=current_active_user#get-the-current-active-user): 
```
optional: If True, None is returned if there is no authenticated user or if it doesn't pass the other requirements. Otherwise, throw 401 Unauthorized. Defaults to False.

active: If True, throw 401 Unauthorized if the authenticated user is inactive. Defaults to False.
```

**Comment by frankie567** [comment, marked as answer]
The `active` parameter will check if the `is_active` property of the user is true or false. When `is_active` if `False` for a user, they can't log in or reset their password. It's a way to disable a user account.

The `optional` parameter is completely different: it's is useful if you need to define routes that *optionally* need a user. For example, imagine an homepage accessible by any user. We need to render the route without error even if there is no authenticated user. However, if a user is logged in, we may want to show a user profile menu.
